vllm/distributed/parallel_state.py

```diff
@@ -162,12 +176,31 @@ def initialize_model_parallel(
             _TP_DEVICE_GROUP = group
             _TP_CPU_GROUP = cpu_group
 
+    # A small all_reduce for warmup.
+    # this is needed because device communicators might be created lazily
+    # (e.g. NCCL). This will ensure that the communicator is initialized
+    # before any communication happens, so that this group can be used for
+    # graph capture immediately.
+    data = torch.zeros(1)
+    if torch.cuda.is_available():
+        data = data.to(device=f"cuda:{_LOCAL_RANK}")
+    torch.distributed.all_reduce(data, group=_TP_DEVICE_GROUP)
```

Comment:
add torch.cuda.sync after every warmup

Reply:
fixed in [8fd22fd](https://github.com/vllm-project/vllm/pull/4754/commits/8fd22fd2d29b52038c6c41f23b2d70894d051fc2).